### PR TITLE
fix(937): add start all child pipelines endpoint

### PR DESCRIPTION
**Pull request by minz1027**
- add start all child pipelines endpoint
- allow to list pipelines by `configPipelineId`

Related to: https://github.com/screwdriver-cd/screwdriver/issues/937

### CODE REVIEW

File: plugins/pipelines/startAll.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const boom = require('boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const idSchema = joi.reach(schema.models.pipeline.base, 'id');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/pipelines/{id}/startall',
```
Comment (tkyi) [inline plugins/pipelines/startAll.js]: camel case maybe? `startAll`?

Comment (minz1027) [reply plugins/pipelines/startAll.js]: don't think it's good to have camel case in url...

Comment (tkyi) [reply plugins/pipelines/startAll.js]: good point whoops

File: plugins/pipelines/startAll.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const pipelineFactory = request.server.app.pipelineFactory;
```
Comment (tkyi) [inline plugins/pipelines/startAll.js]: can do `const { pipelineFactory, eventFactory, userFactory } = request.server.app;`

File: plugins/pipelines/startAll.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            const id = request.params.id;
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+            const scm = pipelineFactory.scm;
+
+            return Promise.all([
+                pipelineFactory.get(id),
+                userFactory.get({ username, scmContext })
+            ])
+                .then(([pipeline, user]) => {
```
Comment (tkyi) [inline plugins/pipelines/startAll.js]: do we need to check if user exists?
```
if (!user) {
     throw boom.notFound(`User ${username} does not exist`);
}
```

File: plugins/pipelines/startAll.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    if (!pipeline) {
+                        throw boom.notFound('Pipeline does not exist');
+                    }
+
+                    return user.getPermissions(pipeline.scmUri)
+                        // check if user has push access
+                        .then((permissions) => {
```
Comment (tkyi) [inline plugins/pipelines/startAll.js]: Are we going to check if the user has permission on the child pipelines? Or is it implied?

Comment (minz1027) [reply plugins/pipelines/startAll.js]: Don't think we need to check here, it is shared permissions.  e.g. I have permissions for child A, B, you have permission for child C and added that. All child A, B, C should start.

File: plugins/pipelines/startAll.js (hunk 3 above)
Comment (minz1027) [reply plugins/pipelines/startAll.js]: We have that in a lot of our endpoints, but i forgot why we need it.. We're getting `username` and `scmContext` from user's JWT, how is it possible that the user doesn't exists?

Comment (tkyi) [reply plugins/pipelines/startAll.js]: I see what you're saying. Should be okay to not have this check then.

File: plugins/pipelines/startAll.js (hunk 4 above)
Comment (tkyi) [reply plugins/pipelines/startAll.js]: I see, so for startall we only care if the user has permissions against the config pipeline. If we did a Start on a childPipeline, we would still check permissions against that child repo.

**Pull request merged**